### PR TITLE
Refactor asset creation

### PR DESCRIPTION
**Pull request by MakMuftic**
Currently, it is just calling the RPC method for creating a new asset and not checking existing assets. 

I am investigating how to fetch existing assets. I see there is this [proposal ](https://ethereum-magicians.org/t/eip-2256-add-wallet-getownedassets-json-rpc-method/3600), but not sure how to fetch assets.

### CODE REVIEW

File: example/src/containers/MetaMaskConnector/MetaMaskConnector.tsx [hunk 1]
```diff
@@ -13,6 +14,10 @@ export const MetaMaskConnector = () => {
        if(!isInitiated) {
            alert("Failed to install snap");
        } else {
+           const dotAssetAdded = await addDotAsset();
```
Comment (MakMuftic) [inline example/src/containers/MetaMaskConnector/MetaMaskConnector.tsx]: @mpetrunic not sure is this right spot in code to add asset ?

Comment (mpetrunic) [reply example/src/containers/MetaMaskConnector/MetaMaskConnector.tsx]: Will this trigger some user prompt?

Comment (MakMuftic) [reply example/src/containers/MetaMaskConnector/MetaMaskConnector.tsx]: @mpetrunic It won't trigger prompt.

**Comment by MakMuftic** [comment]
@mpetrunic I solved asset duplication for now by deleting asset before adding a new one. Still investigating how to solve it properly. Tried reducing asset description object but as mentioned down in documentation it is not really possible.

image

**Pull request merged**